org.eclipse.buildship.core.test/src/main/groovy/org/eclipse/buildship/core/internal/workspace/ImportingWtpProjects.groovy

```diff
@@ -62,17 +61,17 @@ class ImportingWtpProjects extends ProjectSynchronizationSpecification {
                 }
             """
         }
-
-        WorkspaceOperations operations = new WorkspaceOperationsDelegate(wtpInstalled: true)
-        registerService(WorkspaceOperations, operations)
+        wtpinstalled = true
 
         when:
         SynchronizationResult result = tryImportAndWait(root)
+        IProject project = findProject('project')
 
         then:
-        result.status.severity == IStatus.WARNING
-        result.status.exception instanceof UnsupportedConfigurationException
-        result.status.exception.message == "WTP currently does not support mixed deployment paths."
+        result.status.severity == IStatus.OK
```

Comment:
❌ Shouldn't be OK if there were errors.

Reply:
How should we solve this? This change is there because configurators are now responsible for handling their own errors. I guess we should introduce a `ProjectConfiguratorResult` and use it to compute the actual result. WDYT?

Reply:
The context objects we pass to the configurators should have an error-handling method. It should take a message and optionally a throwable and 

- add a Gradle error marker at the most specific location
- add the problem to the aggregate Status we return at the end of synchronization

Reply:
We don't want a return object, because we don't want contributors to instantiate stuff themselves. Also we may want additional stuff returned in the future. Using a single context object for all interactions is the most extensible way to write an SPI.

Reply:
Makes sense, thanks for he elaboration!